### PR TITLE
feat(search-api): #1414 Add new aggregations to dataset search: distO…

### PR DESCRIPTION
**Pull request by markkopaas**
…nNonPublicAccessCount,nonDistOnPublicAccessCount,nonDistOnNonPublicAccessCount

Jira issue link: [link](LINK_HERE)

> check applicable boxes

- [ ] I have made tests to match the user stories
- [ ] This code does not require tests that match user stories

### CODE REVIEW

File: applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java [hunk 1]
```diff
@@ -345,8 +345,31 @@ public SearchRequestBuilder addAggregations(SearchRequestBuilder searchBuilder,
                     .must(QueryBuilders.termQuery("accessRights.code.raw", "PUBLIC"))
             ));
         }
-        if (selectedAggregationFields.contains("subjectCount")) {
+        if (selectedAggregationFields.contains("distOnNonPublicAccessCount")) {
```
Comment (thomasnokleby) [inline applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java]: Hva står distOnNonPublic for  ?

File: applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java [hunk 2]
```diff
@@ -345,8 +345,31 @@ public SearchRequestBuilder addAggregations(SearchRequestBuilder searchBuilder,
...
+
+            searchBuilder.addAggregation(AggregationBuilders.filter("distOnNonPublicAccessCount",
+                QueryBuilders.boolQuery()
+                    .must(QueryBuilders.existsQuery("distribution"))
+                    .mustNot(QueryBuilders.termQuery("accessRights.code.raw", "PUBLIC"))
+            ));
+        }
+        if (selectedAggregationFields.contains("nonDistOnPublicAccessCount")) {
```
Comment (thomasnokleby) [inline applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java]: Hva står nonDistOn for  ?

File: applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java [hunk 3]
```diff
@@ -345,8 +345,31 @@ public SearchRequestBuilder addAggregations(SearchRequestBuilder searchBuilder,
...
 
+            searchBuilder.addAggregation(AggregationBuilders.filter("nonDistOnPublicAccessCount",
+                QueryBuilders.boolQuery()
+                    .mustNot(QueryBuilders.existsQuery("distribution"))
+                    .must(QueryBuilders.termQuery("accessRights.code.raw", "PUBLIC"))
+            ));
+        }
+        if (selectedAggregationFields.contains("nonDistOnNonPublicAccessCount")) {
```
Comment (thomasnokleby) [inline applications/search-api/src/main/java/no/fdk/searchapi/controller/DatasetsSearchController.java]: Som over

**Review by thomasnokleby — CHANGES_REQUESTED** [review]
Trenger enten en forklarende kommentar eller et mindre forkortet variabelnavn.

**Review by oyvindis — APPROVED** [review]

**Review by thomasnokleby — APPROVED** [review]
Ser bra ut!

**Pull request merged**